File: C/Algorithms/1455.c
int isPrefixOfWord(char *sentence, char *searchWord) {
  int idx = 1, try = 1;
  char *runner = searchWord;
  while (*sentence) {
    while (*sentence && *runner && try) {
      if (*sentence != *runner) {
        try = 0;
      } else {
        sentence++, runner++;
      }
    }
    if (!*runner)
      return idx;
    // advance index and make runner
    // point to start of searchWord again.
    if (*sentence == ' ') {
      idx++;
      runner = searchWord;
      try = 1;
    }
    sentence++;
  }
  return -1;
}

File: Rust/Algorithms/1455.rs
impl Solution {
    pub fn is_prefix_of_word(
        sentence: String,
        search_word: String,
    ) -> i32 {
        for (idx, str) in sentence.split(" ").enumerate() {
            if str.starts_with(&search_word) {
                return (idx + 1) as i32;
            }
        }
        -1
    }
}
